akka-actor/src/main/scala/akka/actor/Props.scala

```diff
@@ -130,6 +131,21 @@ object Props {
   def create[T <: Actor](actorClass: Class[T], creator: Creator[T]): Props = {
     apply(defaultDeploy, classOf[CreatorConsumer], actorClass :: creator :: Nil)
   }
+
+  private final def isNotClosingOverEnclosingClass(clazz: Class[_]): Boolean = {
+    def findMatchingConstructor(argTypes: Array[Class[_]]): Boolean =
+      try { val c = clazz.getConstructor(argTypes: _*); true }
+      catch { case ex: NoSuchMethodException ⇒ false }
+    def findMatchingDeclaredConstructor(argTypes: Array[Class[_]]): Boolean =
+      try { val c = clazz.getDeclaredConstructor(argTypes: _*); true }
+      catch { case ex: NoSuchMethodException ⇒ false }
+
+    //noinspection EmptyCheck
+    def defaultEmptyConstructor = clazz.getConstructors.length == 0 && findMatchingDeclaredConstructor(Array.empty)
+    def matchingConstructor = clazz.getDeclaredConstructors.exists { c ⇒ findMatchingConstructor(c.getParameterTypes) }
```

Comment:
Hm... I thought I posted a benchmark together with this PR, seems not or it was lost in a commit-line comment maybe. I remember it not being significant (entire method taking 0.2 micros), but I'll check again - esp. with validating Rolands latest hint